### PR TITLE
[Filtering] Deduplicate semantic duplicate errors

### PR DESCRIPTION
**Pull request by timgrein**
## Closes https://github.com/elastic/enterprise-search-team/issues/6062

The `FilterValidationError` API allows for multiple `ids`. I've changed the API of `SyncRuleValidationResult` to allow to also reference multiple ids as one result could correlate to multiple sync rules. I've adapted the constructor of `SyncRuleValidationResult` that it can still work with a single value (but now also with a list).

This enables the duplication of the semantic duplicate errors messages:

**Note:** _The ids in the box heading require a Kibana change. This change adapts the message inside the box details._

**Before** (two times the same error just flipped): 
image

**After**:
image

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/filtering/validation.py [hunk 1]
```diff
@@ -28,15 +29,21 @@ class SyncRuleValidationResult:
...
     ADVANCED_RULES = "advanced_snippet"
 
-    def __init__(self, rule_id, is_valid, validation_message):
-        self.rule_id = rule_id
+    def __init__(self, rule_ids, is_valid, validation_message):
+        # `str` and `bytes` are also iterable
+        if not isinstance(rule_ids, Iterable) or isinstance(rule_ids, (str, bytes)):
+            rule_ids = [rule_ids]
+        else:
+            rule_ids = list(rule_ids)
```
Comment (timgrein) [inline connectors/filtering/validation.py:37]: I think this is okish for now, as we sometimes use the `rule_ids` keyword explicitly and sometimes we omit it. So this makes sure that everything is working without scanning through every usage of `SyncRuleValidationResult`. Maybe we should revisit using type hints again.

**Comment by artem-shelkovnikov** [comment]
Visually the change still looks non-actionable, is there some follow-up planned for this change?

**Comment by artem-shelkovnikov** [comment]
Ah I see, there's now one message instead of two, which is nice. We can address the readability of it separately, I guess?

**Comment by timgrein** [comment]
> Ah I see, there's now one message instead of two, which is nice. We can address the readability of it separately, I guess?

Yes exactly. The heading is set within Kibana, that's a separate change. The message can be adjusted. WDYT about `Basic rule 1 (id: xyz) is semantically equivalent to Basic rule 2 (id: abc). Please delete one of those rules.`?

**Comment by artem-shelkovnikov** [comment]
> WDYT about Basic rule 1 (id: xyz) is semantically equivalent to Basic rule 2 (id: abc). Please delete one of those rules.?

I think ids are not very useful - they are not exposed in UI at all. Is it possible to instead specify the rule - e.g. `The rule #1 is semantically equal to rule #4` and have #1 and #4 displayed in UI somehow (like the first column, for example)

File: config.yml [hunk 2]
```diff
@@ -14,7 +14,14 @@
...
 ##        api_key: changeme # The Elasticsearch API key used to write data into the content index.
-#connectors: []
+connectors:
+-
+  connector_id: "BdDmZYsBMwlGH8qyGO3r"
+  service_type: "mongodb"
+  api_key: "QjlEbVpZc0JNd2xHSDhxeUx1MVc6aXBubndhc1dScC1lbWVYTndZbWFjZw=="
+elasticsearch:
+  host: "http://localhost:9200"
+  api_key: "QjlEbVpZc0JNd2xHSDhxeUx1MVc6aXBubndhc1dScC1lbWVYTndZbWFjZw=="
```
Comment (seanstory) [inline config.yml:24]: I expect you didn't mean to to commit this

**Comment by artem-shelkovnikov** [comment]
@timgrein do you still want to pursue these changes?

**Comment by seanstory** [comment]
Bump @timgrein can you either close this or get it cleaned up and ready for re-review?

**Comment by timgrein** [comment]
Can be closed, there's probably a better solution for this issue

**Pull request closed without merging**